scripts/feature_tests/upgrade/common.sh

```diff
@@ -0,0 +1,486 @@
+    scale_to="${1}"
+    cp_node_ip="${2}"
+    node_type="${3:-master}"
+    replicas=0
+
+    echo "Scaling ${node_type} nodes to ${scale_to}"
+    if [[ "${node_type}" == "master" ]]; then
+        kubectl get kcp -n metal3 test1 -o json | jq '.spec.replicas='"${scale_to}"'' | kubectl apply -f-
+    else
+        kubectl get machinedeployment -n metal3 test1 -o json | jq '.spec.replicas='"${scale_to}"'' | kubectl apply -f-
+    fi
+
+    for i in {1..1800};do
+        if [[ "${node_type}" == "master" ]]; then
+            replicas=$(ssh "-o LogLevel=quiet" -o "UserKnownHostsFile=/dev/null" -o PasswordAuthentication=no -o "StrictHostKeyChecking no" "${UPGRADE_USER}@${cp_node_ip}" -- kubectl get nodes| grep -c master)
+        else
+            replicas=$(ssh "-o LogLevel=quiet" -o "UserKnownHostsFile=/dev/null" -o PasswordAuthentication=no -o "StrictHostKeyChecking no" "${UPGRADE_USER}@${cp_node_ip}" -- kubectl get nodes | awk 'NR>1' | grep -cv master)
+        fi
+
+        if [[ "$replicas" == "${scale_to}" ]]; then
+            echo ''
+            echo "Successfully scaled ${node_type} nodes to ${scale_to}"
+            break
+        fi
+        # Scaling progress indicator
+        echo -n "*"
+        sleep 10
+        if [[ "${i}" -ge 1800 ]];then
+            echo "Error: Scaling of ${node_type} nodes took to long"
+            deprovision_cluster
+            wait_for_cluster_deprovisioned
+            break
+        fi
+    done
+}
+
+function deploy_workload_on_workers {
+echo "Deploy workloads on workers"
+# Deploy workloads
+cat <<EOF | kubectl apply -f -
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: workload-1-deployment
+spec:
+  replicas: 10
+  selector:
+    matchLabels:
+      app: workload-1
+  template:
+    metadata:
+      labels:
+        app: workload-1
+    spec:
+      containers:
+      - name: nginx
+        image: nginx
+EOF
+
+echo "Waiting for workloads to be ready"
+for i in {1..1800};do
+    workload_replicas=$(kubectl get deployments workload-1-deployment -o json | jq '.status.readyReplicas')
+    if [[ "$workload_replicas" == "10" ]]; then
+        echo ''
+        echo "Successfully deployed workloads across the cluster"
+        break
+    fi
+    echo -n "*"
+    sleep 10
+    if [[ "${i}" -ge 1800 ]];then
+        echo "Error: Workload failed to be deployed on the cluster"
+        deprovision_cluster
+        wait_for_cluster_deprovisioned
+        break
+    fi
+done
+}
+
+function manage_node_taints {
+    # Enable workload on masters
+    ssh -o PasswordAuthentication=no -o "StrictHostKeyChecking no" "${UPGRADE_USER}@${1}" -- sudo apt-get install jq -y
+    # kubectl get nodes -o json | jq ".items[]|{name:.metadata.name, taints:.spec.taints}"
+    # untaint all masters (one workers also gets untainted, doesn't matter):
+    ssh -o PasswordAuthentication=no -o "StrictHostKeyChecking no" "${UPGRADE_USER}@${1}" -- kubectl taint nodes --all node-role.kubernetes.io/master-
```

Comment:
add a line break please. And you could gather the repeating options in a variable